plugins/inputs/prometheus/prometheus.go

```diff
@@ -77,6 +93,14 @@ func init() {
 		return &Prometheus{
 			mbCh:         make(chan PrometheusMetricBatch, 10000),
 			shutDownChan: make(chan interface{}),
+			middleware: agenthealth.NewAgentHealth(
+				zap.NewNop(),
```

Comment:
doens't this mean we won't be getting any logs related to agent health?

Reply:
Yes this will only attach the status code handler. We aren't trying to get any other logs related to agent health in this pr.